diff -Naur old/nfd_gtk.c new/nfd_gtk.c
--- old/nfd_gtk.c	2018-05-02 18:21:42.000000000 +0200
+++ new/nfd_gtk.c	2018-05-02 18:26:29.000000000 +0200
@@ -4,6 +4,9 @@
   http://www.frogtoss.com/labs
 */
 
+/* Linux only */
+#ifdef NATIVE_BUILD
+
 #include <stdio.h>
 #include <assert.h>
 #include <string.h>
@@ -25,7 +28,7 @@
         strncat( filterName, SEP, bufsize - len - 1 );
         len += strlen(SEP);
     }
-    
+
     strncat( filterName, typebuf, bufsize - len - 1 );
 }
 
@@ -118,7 +121,8 @@
     pathSet->count = (size_t)g_slist_length( fileList );
     assert( pathSet->count > 0 );
 
-    pathSet->indices = NFDi_Malloc( sizeof(size_t)*pathSet->count );
+    pathSet->indices = NFDi_Malloc( sizeof((size_t*)pathSet->count) );
+
     if ( !pathSet->indices )
     {
         return NFD_ERROR;
@@ -153,7 +157,7 @@
     }
 
     g_slist_free( fileList );
-    
+
     return NFD_OKAY;
 }
 
@@ -165,10 +169,10 @@
                                  
 /* public */
 
-nfdresult_t NFD_OpenDialog( const char *filterList,
+nfdresult_t NFD_OpenDialog( const nfdchar_t *filterList,
                             const nfdchar_t *defaultPath,
                             nfdchar_t **outPath )
-{    
+{
     GtkWidget *dialog;
     nfdresult_t result;
 
@@ -196,6 +200,7 @@
     {
         char *filename;
 
+
         filename = gtk_file_chooser_get_filename( GTK_FILE_CHOOSER(dialog) );
 
         {
@@ -377,3 +382,5 @@
     
     return result;
 }
+
+#endif /* NATIVE_BUILD */
diff -Naur old/nfd.h new/nfd.h
--- old/nfd.h	2018-05-02 18:21:42.000000000 +0200
+++ new/nfd.h	2017-10-16 07:12:26.000000000 +0200
@@ -7,6 +7,12 @@
  */
 
 
+/* 
+   To be tested:
+   https://github.com/mlabbe/nativefiledialog/pull/33/commits/dedf95faa754b00a34c13bc4cf587e16d002b67c
+
+*/
+
 #ifndef _NFD_H
 #define _NFD_H
 
diff -Naur old/nfd_win.cpp new/nfd_win.cpp
--- old/nfd_win.cpp	2018-05-02 18:21:42.000000000 +0200
+++ new/nfd_win.cpp	2017-11-24 00:24:57.000000000 +0100
@@ -4,6 +4,8 @@
   http://www.frogtoss.com/labs
  */
 
+#ifdef _WIN32
+
 /* only locally define UNICODE in this compilation unit */
 #ifndef UNICODE
 #define UNICODE
@@ -17,19 +19,26 @@
 
 #include <wchar.h>
 #include <stdio.h>
+#include <iostream>
 #include <assert.h>
 #include <windows.h>
+#ifdef _WIN32
+#include <shobjidl.h>
+#else
 #include <ShObjIdl.h>
+#endif
+#include "nfd.h"
 #include "nfd_common.h"
 
 
+
 // allocs the space in outPath -- call free()
 static void CopyWCharToNFDChar( const wchar_t *inStr, nfdchar_t **outStr )
 {
     int inStrCharacterCount = static_cast<int>(wcslen(inStr)); 
     int bytesNeeded = WideCharToMultiByte( CP_UTF8, 0,
                                            inStr, inStrCharacterCount,
-                                           NULL, 0, NULL, NULL );    
+                                           NULL, 0, NULL, NULL );
     assert( bytesNeeded );
     bytesNeeded += 1;
 
@@ -83,10 +92,10 @@
     assert( charsNeeded );
     assert( !*outStr );
     charsNeeded += 1; // terminator
-    
-    *outStr = (wchar_t*)NFDi_Malloc( charsNeeded * sizeof(wchar_t) );    
+
+    *outStr = (wchar_t*)NFDi_Malloc( charsNeeded * sizeof(wchar_t) );
     if ( !*outStr )
-        return;        
+        return;
 
     int ret = MultiByteToWideChar(CP_UTF8, 0,
                                   inStr, inStrByteCount,
@@ -107,7 +116,7 @@
 {
     const char SEP[] = ";";
     assert( specBufLen > strlen(ext)+3 );
-    
+
     if ( strlen(specBuf) > 0 )
     {
         strncat( specBuf, SEP, specBufLen - strlen(specBuf) - 1 );
@@ -116,8 +125,15 @@
 
     char extWildcard[NFD_MAX_STRLEN];
     int bytesWritten = sprintf_s( extWildcard, NFD_MAX_STRLEN, "*.%s", ext );
-    assert( bytesWritten == strlen(ext)+2 );
-    
+
+    std::cout << "extWildcard : " << extWildcard << std::endl;
+    std::cout << "NFD_MAX_STRLEN : " << NFD_MAX_STRLEN << std::endl;
+    std::cout << "ext : " << ext << std::endl;
+    std::cout << "bytesWritten : " << bytesWritten << std::endl;
+    std::cout << "strlen(ext)+2 : " << strlen(ext)+2 << std::endl;
+
+    // empty filter ?
+    //assert( bytesWritten == strlen(ext)+2 );
     strncat( specBuf, extWildcard, specBufLen - strlen(specBuf) - 1 );
 
     return NFD_OKAY;
@@ -352,8 +368,8 @@
 
 /* public */
 
-
-nfdresult_t NFD_OpenDialog( const char *filterList,
+#ifdef DO_NOT_USE
+nfdresult_t NFD_OpenDialog( const nfdchar_t *filterList,
                             const nfdchar_t *defaultPath,
                             nfdchar_t **outPath )
 {
@@ -534,6 +550,7 @@
     return nfdResult;
 }
 
+
 nfdresult_t NFD_SaveDialog( const nfdchar_t *filterList,
                             const nfdchar_t *defaultPath,
                             nfdchar_t **outPath )
@@ -622,6 +639,8 @@
     return nfdResult;
 }
 
+#endif
+
 class AutoCoInit
 {
 public:
@@ -751,3 +770,5 @@
 
     return NFD_OKAY;
 }
+
+#endif /* __Linux */
